### PR TITLE
[BUGFIX] Load layer from definition does not use qlr as relative path

### PR DESCRIPTION
**Pull request by rldhont**
Fixes #15656 Loading QLR file with relative path does not work correctly

Has been fixed in master but was not fixed in LTRs.

## Checklist
- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/core/qgsproject.h [hunk 1]
```diff
@@ -259,7 +259,7 @@ class CORE_EXPORT QgsProject : public QObject
     QString writePath( const QString& filename, const QString& relativeBasePath = QString::null ) const;
 
     /** Turn filename read from the project file to an absolute path */
-    QString readPath( QString filename ) const;
+    QString readPath( QString filename, const QString& relativeBasePath = QString::null ) const;
```
Comment (m-kuhn) [inline src/core/qgsproject.h]: s/QString::null/QString()/

Comment (rldhont) [reply src/core/qgsproject.h]: Thnaks for the review, I'll fix it

**Review by NathanW2 — APPROVED** [review]
@rldhont Looks good to me but I don't have a way to test currently but I trust you :)

**Comment by NathanW2** [comment]
@rldhont I'm ok for you to merge this.

**Comment by rldhont** [comment]
thanks @NathanW2 !

**Comment by NicolaiMogensen** [comment]
I suspect that this breaks backwards compability. 

Qgis >=2.18.15:
`static bool loadLayerDefinition( QDomDocument doc, QgsLayerTreeGroup* rootGroup, QString &errorMessage, const QString& relativeBasePath = QString() );`

Qgis < 2.18.15:
`static bool loadLayerDefinition( QDomDocument doc, QgsLayerTreeGroup* rootGroup, QString &errorMessage /Out/ );`

Notice the /Out/ tag is missing now. 

The added QString argument throws not enough arguments for code written for qgis < 2.18.15. If you add the QString it will work for qgis >= 2.18.15 but throw a too many arguments for qgis < 2.18.15.

**Pull request merged**